docs/rsdl/rapid-rsdl-capability-mapping.mdx

```diff
@@ -0,0 +1,221 @@
+---
+id: rsdl-capability-mapping
+title: Capability semantics
+sidebar_label: RSDL API Capability semantics
+---
+
+> DRAFT
+> Initial Draft. Dec 2021
+
+Mapping RSDL capabilities to CSDL annotations
+
+## Delete
+
+```rsdl
+path /skus/{id} {
+    DELETE
+}
+```
+
+```xml
+ <Annotations Target="skus">
+  <Annotation Term="Org.OData.Capabilities.V1.DeleteRestrictions">
+    <Record Type="Org.OData.Capabilities.V1.DeleteRestrictionsType">
+      <PropertyValue Property="Deletable" Bool="true" />
+    </Record>
+  </Annotation>
+</Annotations>
+```
+
+## Update
+
+```rsdl
+path /skus/{id} {
+    PATCH {
+        excluded { id releaseDate }
+    }
+}
+```
+
+```xml
+<Annotations Target="skus">
+  <Annotation Term="Org.OData.Capabilities.V1.UpdateRestrictions">
+    <Record Type="Org.OData.Capabilities.V1.UpdateRestrictionsType">
+      <PropertyValue Property="Updatable" Bool="true" />
+      <PropertyValue Property="NonUpdatableProperties">
+        <Collection>
+          <PropertyPath>id</PropertyPath>
+          <PropertyPath>releaseDate</PropertyPath>
+        </Collection>
+      </PropertyValue>
+    </Record>
+  </Annotation>
+</Annotations>
+```
+
+## Insert
+
+```rsdl
+path /orders {
+    POST {
+        required { externalId }
+        excluded { id orderDate }
+    }
+}
+```
+
+```json
+{
+  "$Annotations": {
+    "orders": {
+      "@Org.OData.Capabilities.V1.InsertRestrictions": {
+        "$Type": "Org.OData.Capabilities.V1.InsertRestrictionsType",
+        "Insertable": true,
+        "RequiredProperties": ["id", "orderDate"],
+        "NonInsertableProperties": ["id", "orderDate"]
+      }
+    }
+  }
+}
+```
+
+```xml
+<Annotations Target="orders">
+  <Annotation Term="Org.OData.Capabilities.V1.InsertRestrictions">
+    <Record Type="Org.OData.Capabilities.V1.InsertRestrictionsType">
+      <PropertyValue Property="Insertable" Bool="true" />
+      <PropertyValue Property="RequiredProperties">
+        <Collection>
+          <PropertyPath>id</PropertyPath>
+          <PropertyPath>orderDate</PropertyPath>
+        </Collection>
+      </PropertyValue>
+      <PropertyValue Property="NonInsertableProperties">
+        <Collection>
+          <PropertyPath>id</PropertyPath>
+          <PropertyPath>orderDate</PropertyPath>
+        </Collection>
+      </PropertyValue>
+    </Record>
+  </Annotation>
+</Annotations>
+```
+
+## Paging
+
+```rsdl
+path /skus {
+    GET {
+        paging
+    }
+}
+```
+
+```xml
+<Annotations Target="skus">
+  <Annotation Term="Org.OData.Capabilities.V1.TopSupported" Bool="true" />
+  <Annotation Term="Org.OData.Capabilities.V1.SkipSupported" Bool="true" />
+</Annotations>
+```
```

Comment:
Again: this is the default, need to opt out of paging

Reply:
I need to think a little but about generating this (or to be precise generating the `false` for the GET requests that do not contain `paging`